src/app/shared/widgets/location-dropdowns/location-dropdowns.component.ts

```diff
@@ -0,0 +1,135 @@
+import { FormControl, FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { CommunityService } from '../../../_services/api/community.service';
+import { CountryService } from '../../../_services/api/country.service';
+import { DistrictService } from '../../../_services/api/district.service';
+import { RegionService } from '../../../_services/api/region.service';
+import { FormActionsService } from '../../../_services/form-actions.service';
+import { NotificationService } from '../../../_services/notification.service';
+import { FormElementBase } from '../../dynamic-form/types/form-element-base';
+
+@Component({
+  selector: 'app-location-dropdowns',
+  templateUrl: './location-dropdowns.component.html',
+  styleUrls: ['./location-dropdowns.component.scss'],
+})
+export class LocationDropdownsComponent implements OnInit, OnDestroy {
+  config: FormElementBase<string>;
+  group: FormGroup;
+
+  data: any;
+  fields: string[] = [];
+  locationForm = new FormGroup({});
+
+  subscriptions: Subscription[] = [];
+
+  constructor(
+    public countryService: CountryService,
+    public regionService: RegionService,
+    public districtService: DistrictService,
+    public communityService: CommunityService,
+    private notificationService: NotificationService,
+    private formActionService: FormActionsService
+  ) {}
+
+  ngOnInit(): void {
+    this.data = this.config?.widgetInfo || {};
+    this.fields = Object.keys(this.data);
+    this.createForm();
+    this.populateOptions();
+  }
+
+  createForm(): void {
+    this.fields.forEach((field) => {
+      const originalControl = this.group.controls[this.data[field].key];
+      this.locationForm.addControl(field, new FormControl(originalControl?.value));
+      if (this.data[field].disabled) {
+        this.locationForm.controls[field].disable();
+      }
+      const control = this.locationForm.get(field);
+      if (control) {
+        this.subscriptions.push(
+          control.valueChanges.subscribe((val: any) => {
+            originalControl?.setValue(val);
+            this.formActionService.setInputChange(this.data[field].key, true);
+            this.updateDependents(field, val);
+          })
+        );
+      }
+    });
+    this.subscriptions.push(this.formActionService.getDiscard().subscribe(() => this.resetForm()));
+  }
+
+  resetForm(): void {
+    this.fields.forEach((field) => {
+      const originalControl = this.group.controls[this.data[field].key];
+      this.data[field].options = [];
+      this.locationForm.get(field)?.setValue(originalControl?.value, { emitEvent: false });
+    });
+    this.populateOptions();
+  }
+
+  populateOptions(): void {
+    this.fields.forEach((field) => {
+      const control = this.locationForm.controls[field];
+      if (!control) {
+        return;
+      }
+      const dependeeControl = this.locationForm.controls[this.data[field].dependingOn];
+      if (dependeeControl?.value || field === this.fields[0]) {
```

Comment:
while this should work, I think we can generalize a bit here and have something like:
if (!dependeeControl || dependeeControl.value)
that way we can have all the fields independent (for example we could have both region and district in the form with all the options from the start - like we need on the stats page)

Reply:
Changed, thx!